Fix error when using with webpack ('stream is not defined'), fixes #5

beeman opened the pull request:
Fixes #5

slnode commented:
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

slnode commented:
Can one of the admins verify this patch?

beeman commented:
Any updates on this PR?

slnode commented:
Can one of the admins verify this patch?

slnode commented:
Can one of the admins verify this patch?

davidcheung commented:
@slnode ok to test

davidcheung commented:
@beeman can you please rebase the code on top of master branch and make sure the tests are passing

beeman commented:
@davidcheung I've rebased the code and fixed the commit check.

About the error in `npm test`, it seems that the code needs to be `browserified` in order to pass the test. It looks like this also happend with the file `modules/change-stream/stream.js` before it got added to the project.

I am using this branch in a project that gets compiled with webpack so I don't need it.

The best solution is probably to add a browserify step to gulp that compiles it to a backwards-compatible format, though I'm not sure it's worth the effort.

davidcheung commented:
yeah dropping `browserify` for something else would definitely bigger than just an `angular-live-set` decision.  Hmm perhaps we could create an issue here as a `feature request` to support web-pack

davidcheung commented:
@bajtos any suggestions how to achieve compatibility with both? 
@beeman have you tried if @ritch's suggestion https://github.com/strongloop/angular-live-set/issues/5#issuecomment-132769024 works?

bajtos commented on modules/change-stream/change-stream.js:
```diff
@@ -5,6 +5,7 @@
 
 angular.module('ls.ChangeStream',[]).factory('createChangeStream', ['$rootScope', function($rootScope) {
   function createChangeStream(eventSource, $scope) {
+    var stream = require('stream')
```
We usually place require()s at the top of the file:

```js
var stream = require('stream');

angular.module(//...
```

davidcheung replied:
our test phantomjs browser doenst have requirejs or such package managers, so it cant do `require('stream')`

beeman replied:
This is an issue with the file `change-stream.js` because it's not being prepared for being loaded in the browser, as opposed to `stream.js` in the same folder (the file that's being included with the `require` statement. The latter file seems to be browserified which makes it that it works in browsers.

I've tried running the `browserify` command on `change-stream.js` and that makes it work, but it also makes the file huge and adds a lot of cruft.

That's why it would make sense to have this done in a build-step, also before running the tests in PhantomJS.

bajtos commented:
@ritch you are the original author of this module IIRC. Could you please take a look at this pull request and advise us what's the best way for fixing the problem?

bajtos commented:
Here is my understanding: the current version of the module is designed to be loaded directly from the HTML page and does not support browserify/webpack.

The change presented in this pull request may fix the problem for browserify/webpack, but it breaks the ability to load the module directly.

I think the correct approach is to implement the proposal from https://github.com/strongloop/angular-live-set/issues/5#issuecomment-132769024.
